.github/workflows/PR-build.yml

```diff
@@ -0,0 +1,151 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+
+name: PR Build
+on:
+  pull_request:
+    branches: 
+      - main 
+    types: 
+      - opened
+      - synchronize
+      - reopened
+      - ready_for_review
+    paths-ignore:
+      - '**/*.md'
+      - 'NOTICE'
+      - 'RELEASE_NOTES'
+      - 'THIRD-PARTY'
+      - 'LICENSE'
+      - '.github/**'
+      - '!.github/workflows/PR-build'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  macos-unittest:
```

Comment:
Why can't this be done with os list with a single job?

Reply:
You can argue they would have similar workflows. However, there are some workflows that are specific to OS (e.g Windows when install [choco-make](https://github.com/khanhntd/amazon-cloudwatch-agent/blob/master/.github/workflows/build-test-windows.yml#L49-L50)) or having coverage-uploading that is one workflow would be enough. Instead of having multiple `if else`, I would prefer a straightforward workflow. If you have a strong urge against it, I am fine with changing it too